pkg/daemon/controller/controller.go

```diff
@@ -321,8 +321,7 @@ func (sc *ServiceController) reconcileEnvironmentVariables(envVars map[string]st
 
 // ensureVarsAreUpToDate ensures that the proxy environment variables are set as expected on the instance
 // If there's any changes, an instance restart is required to ensure all processes pick up the updated values.
-func (sc *ServiceController) ensureVarsAreUpToDate(envVars map[string]string) (bool, error) {
-	// systemEnvVarRegistryPath is where system level environment variables are stored in the Windows OS
+func ensureVarsAreUpToDate(envVars map[string]string, watchedEnvVars []string) (bool, error) {
 	const systemEnvVarRegistryPath = `SYSTEM\CurrentControlSet\Control\Session Manager\Environment`
```

Comment:
You should use the `SystemEnvVarRegistryPath` here also.

Reply:
this global variable is introduced in  the cleanup commit where it will be used out of this file.